pubsub/subscriptions.js

```diff
@@ -168,8 +188,71 @@ function pullMessages (subscriptionName, callback) {
 }
 // [END pubsub_pull_messages]
 
+let subscribeCounterValue = 1;
+
+function getSubscribeCounterValue () {
+  return subscribeCounterValue;
+}
+
+function setSubscribeCounterValue (value) {
+  subscribeCounterValue = value;
+}
+
+// [START pubsub_pull_ordered_messages]
+const outstandingMessages = {};
+
+function pullOrderedMessages (subscriptionName, callback) {
+  // Instantiates the client library
+  const pubsubClient = PubSub();
+
+  // References an existing subscription, e.g. "my-subscription"
+  const subscription = pubsubClient.subscription(subscriptionName);
+
+  // Pulls messages. Set returnImmediately to false to block until messages are
+  // received.
+  subscription.pull({ returnImmediately: true }, (err, messages) => {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    messages.forEach((message) => {
```

Comment:
Maybe include a comment that mentions that the (re-)ordering is being done by JS, and not by the API itself?